ambari-server/src/main/java/org/apache/ambari/server/api/services/AlertTargetService.java

```diff
@@ -33,54 +33,165 @@
+import io.swagger.annotations.ApiResponses;
+
+
 
 /**
  * The {@link AlertTargetService} handles CRUD operation requests for alert
  * targets.
  */
 @Path("/alert_targets/")
+@Api(value = "/alerts", description = "Endpoint for alert specific operations")
 public class AlertTargetService extends BaseService {
 
-  @GET @ApiIgnore // until documented
+  public static final String ALERT_TARGET_REQUEST_TYPE = "org.apache.ambari.server.controller.AlertTargetSwagger";
+
+  /**
+   * Handles GET /alert_targets
+   * Get all alert targets.
+   *
+   * @param headers http headers
+   * @param ui      uri info
+   * @return alert target resource representation
+   */
+  @GET @ApiIgnore // documented below
   @Produces("text/plain")
   public Response getTargets(@Context HttpHeaders headers,
       @Context UriInfo ui) {
     return handleRequest(headers, null, ui, Request.Type.GET,
         createAlertTargetResource(null));
   }
 
-  @GET @ApiIgnore // until documented
-  @Produces("text/plain")
+  /**
+   * Handles GET /alert_targets/{targetId}
+   * Get a specific alert target.
+   *
+   * @param headers   http headers
+   * @param ui        uri info
+   * @param targetId  alert target id
+   * @return alert target resource representation
+   */
+  @GET
   @Path("{targetId}")
+  @Produces("text/plain")
+  @ApiOperation(value = "Returns a single alert target", response = AlertTargetSwagger.class, responseContainer = "List")
+  @ApiImplicitParams({
+          @ApiImplicitParam(name = QUERY_FIELDS, value = QUERY_FILTER_DESCRIPTION, defaultValue = "AlertTarget/*", dataType = DATA_TYPE_STRING, paramType = PARAM_TYPE_QUERY),
+          @ApiImplicitParam(name = QUERY_SORT, value = QUERY_SORT_DESCRIPTION, dataType = DATA_TYPE_STRING, paramType = PARAM_TYPE_QUERY),
+          @ApiImplicitParam(name = QUERY_PAGE_SIZE, value = QUERY_PAGE_SIZE_DESCRIPTION, defaultValue = DEFAULT_PAGE_SIZE, dataType = DATA_TYPE_INT, paramType = PARAM_TYPE_QUERY),
+          @ApiImplicitParam(name = QUERY_FROM, value = QUERY_FROM_DESCRIPTION, allowableValues = QUERY_FROM_VALUES, defaultValue = DEFAULT_FROM, dataType = DATA_TYPE_INT, paramType = PARAM_TYPE_QUERY),
+          @ApiImplicitParam(name = QUERY_TO, value = QUERY_TO_DESCRIPTION, allowableValues = QUERY_TO_VALUES, dataType = DATA_TYPE_INT, paramType = PARAM_TYPE_QUERY),
+  })
+  @ApiResponses({
+          @ApiResponse(code = HttpStatus.SC_OK, message = MSG_SUCCESSFUL_OPERATION),
+          @ApiResponse(code = HttpStatus.SC_NOT_FOUND, message = MSG_CLUSTER_NOT_FOUND),
+          @ApiResponse(code = HttpStatus.SC_UNAUTHORIZED, message = MSG_NOT_AUTHENTICATED),
+          @ApiResponse(code = HttpStatus.SC_FORBIDDEN, message = MSG_PERMISSION_DENIED),
+          @ApiResponse(code = HttpStatus.SC_INTERNAL_SERVER_ERROR, message = MSG_SERVER_ERROR),
+          @ApiResponse(code = HttpStatus.SC_BAD_REQUEST, message = MSG_INVALID_ARGUMENTS),
+  })
   public Response getTargets(@Context HttpHeaders headers,
```

Comment:
Ideally this method should be `getTarget`, since it returns a single item.  If you want to keep the existing name to avoid mixing documentation and refactoring, you can provide a `nickname` for Swagger.

Reply:
renamed to `getTarget`